### ISSUE TITLE
How do I use pre-trained word vectors?

### ISSUE THREAD

**Issue by glaswasser** [datquocnguyen/LFTM#12]
Hi,
first of all, thanks a lot for your implementation of this method! Unfortunately, I am quite new to Java, so I first tried to run your code on the test example, which worked. 

Now I want to apply topic modelling to my own corpus of 4800 short texts. I have them in the .txt file format with 1 line per text. I named it corpus.txt.

I would like to apply a pre-trained vector from Word2Vec. The file I get from them is a .bin.gz file. However, in your example, you have a .txt file containing the word vectors.

How do I now change this to use the pre-trained vectors? I tried  `java -jar jar/LFTM.jar -model LFLDA -corpus test/corpus.txt -vectors test/knowledge-vectors-skipgram1000.bin -ntopics 4 -alpha 0.1 -beta 0.01 -lambda 0.6 -initers 500 -niters 50 -name testLFLDA`  but this gets me:

`Reading topic modeling corpus: test/corpus.txt
Reading word vectors from word-vectors file test/knowledge-vectors-skipgram1000.bin...
java.lang.ArrayIndexOutOfBoundsException: 1
        at models.LFLDA.readWordVectorsFile(Unknown Source)
        at models.LFLDA.<init>(Unknown Source)
        at LFTM.main(Unknown Source)
The word "ordered" doesn't have a corresponding vector!!!
Error: null
java.lang.Exception
        at models.LFLDA.readWordVectorsFile(Unknown Source)
        at models.LFLDA.<init>(Unknown Source)
        at LFTM.main(Unknown Source)`

The word "ordered" is the first word in my texts, so it does not seem to work at all. Same if I try the `bin.gz` file.

Could you help me out in that matter?
Thanks a lot!

**Comment by datquocnguyen** [comment]
The pre-trained word vector file must be in the .txt file format. See the example file: https://github.com/datquocnguyen/LFTM/blob/master/test/wordVectors.txt

**Comment by glaswasser** [comment]
Okay, I managed to find and download a huge .txt vector file now. How can I remove words from my corpus that don't exist in the vector file? It's way too large to load it into memory :-/

EDIT. I managed to remove the words using python. 
Another question:
In the .topicAssignments document there are the topic assignments for each word in the documents? Why are there 8 different numbers when I only specified 4 topics?
